src/quo/components/token.cljs

```diff
@@ -0,0 +1,53 @@
+(ns quo.components.token
+  (:require [clojure.string :as string]
+            [quo.foundations.resources :as resources]
+            [react-native.core :as rn]
+            [utils.number]))
+
+(defn- size->number
+  "Remove `size-` prefix in size keywords and returns a number useful for styling."
+  [k]
+  (utils.number/parse-int (subs (name k) 5)))
+
+(defn- token-style
+  [style size]
+  (let [size-number (if (keyword? size)
+                      (size->number size)
+                      size)]
+    (assoc style
+           :width  size-number
+           :height size-number)))
+
+(defn- get-token-image*
+  [token]
+  (let [token-name (condp #(%1 %2) token
+                     string?  (keyword (string/lower-case token))
+                     keyword? (keyword (string/lower-case (name token)))
+                     nil)]
+    (resources/get-token token-name)))
+
+(def ^:private get-token-image (memoize get-token-image*))
+(def ^:private b64-png-image-prefix "data:image/png;base64,")
+
+(defn view
+  "Render a token image.
```

Comment:
schema would be great to add here in place of doc string 👍

Reply:
Will add it!

Reply:
@ilmotta are we adding regex validations for all of our string/keyword props?

Sometimes we need it and that's ok, but I'd prefer not using them when it's not required.
A regex is very error prune (do we need tests for them?), IMO they are "kind of" easy to write, but hard to read in the future and also hard to maintain.

If we want to properly validate our strings/keywords, we'd need a more formal specification of what we can/can't receive.

For example, this token schema, what's the minimum length? why 2? what's the maximum? don't we have tokens returned from status-go that contain numbers or other symbols? who is able to answer these kind of questions for all the components we use?

On the other hand, if we create very flexible regexes, does that makes sense? I mean, what would be the benefit from it if the regex is not strict enough?

Reply:
> all of these are valid and resolve to the same

The real issue seems to be why we need to support all these formats.

We can just follow the convention of what a token `symbol` usually looks like, all uppercase, numbers are ok, minimum length is 2, since symbol is actually optional in erc20 and just a string that can contain anything

Reply:
@ulisesmac, the code I suggested was just to help guide the final implementation if and only if you want to add the schemas in this PR. I should've been more clear. It's not something to be copy & pasted.

> @ilmotta are we adding regex validations for all of our string/keyword props?

It depends on how much we want to instrument and how specific we want them to be. There's no right or wrong. Our schemas can be quite dumb, focusing more on primitive types only, like sequences, strings, keywords, etc. But they can be much more useful than that because schemas are runtime constructs, we can check a lot more during instrumentation. More strict schemas are helpful for code generation, something we will eventually take advantage of.

Just a reminder, all the schema code we are writing is dev-only code, it should help us during development. If a schema is too complicated or doesn't help, we can throw away/change.

> On the other hand, if we create very flexible regexes, does that makes sense? I mean, what would be the benefit from it if the regex is not strict enough?

I just tried to convert the docstring you wrote into a valid schema and added a restriction in size to show the utility/power because most devs in the team haven't used malli yet. Don't worry too much about my code suggestion.

> If we want to properly validate our strings/keywords, we'd need a more formal specification of what we can/can't receive.

I agree, it's up to us to define that when working on the PRs. The schemas can get as "formal" or strict as we want them to be. We can start with just `[:or string? keyword?]` and evolve from there.

> A regex is very error prune (do we need tests for them?)

I think you're over generalizing regular expressions. Of course I agree they can be error prone, but the one I shared reads like English for many devs.

We don't need tests for schemas in general, unless they're much more complicated and used by production code. We are far away from that scenario. Instrumentation is dev-only code.

> We can just follow the convention of what a token symbol usually looks like, all uppercase, numbers are ok, minimum length is 2, since symbol is actually optional in erc20 and just a string that can contain anything

@yqrashawn seems reasonable 👍🏼

The fact that we're finally having a discussion about the types and formats of things is a good sign, we are thinking more about the data layer. That's kind of the whole point about using malli :)

Reply:
> > All of these are valid and resolve to the same
> 
> The real issue seems to be why we need to support all these formats.
> 
> We can just follow the convention of what a token `symbol` usually looks like, all uppercase, numbers are ok, minimum length is 2, since symbol is actually optional in erc20 and just a string that can contain anything

@yqrashawn

I'm supporting those formats because the token component is now being used directly or indirectly in many places in the app, replacing code that didn't expect the proper format. 

For example, our current (develop) code to resolve tokens expect a lower case keyword, and the new implementation should break those existing uses.

Regarding to the token's minimum length, I've noticed we have a token image saved as `R`, so our keyword to get it is `:r`.

Seems like I can't be too strict with this schema.

Reply:
@ilmotta

Thanks for the detailed answer, I got the point. 👍